Sources/ArcGISToolkit/Components/Authentication/CertificatePickerViewModifier.swift

```diff
@@ -92,11 +103,41 @@ import ArcGIS
         challenge.resume(with: .cancel)
     }
 
-    private func showCertificateImportError(_ error: Error?) async {
+    private func showCertificateError(_ error: CertificateError) async {
         // This is required to prevent an "already presenting" error.
         try? await Task.sleep(nanoseconds: 100_000)
```

Comment:
I think the proper solution is to annotate this method with `MainActor`.

Reply:
I don't see `already presenting error` with or without `sleep` or `MainActor`. In fact, I see following before this prompt when we ask to input the password.

```
2023-01-23 14:45:44.896226-0800 AuthenticationExample[28329:387920] [DocumentManager] The view service did terminate with error: Error Domain=_UIViewServiceErrorDomain Code=1 "(null)" UserInfo={Terminated=disconnect method}
2023-01-23 14:45:45.005242-0800 AuthenticationExample[28329:387920] [LayoutConstraints] Changing the translatesAutoresizingMaskIntoConstraints property of a UICollectionViewCell that is managed by a UICollectionView is not supported, and will result in incorrect self-sizing. View: <_UIAlertControllerTextFieldViewCollectionCell: 0x13f4083f0; frame = (0 0; 270 24); gestureRecognizers = <NSArray: 0x600001df2520>; layer = <CALayer: 0x6000013beca0>>
2023-01-23 14:45:45.434019-0800 AuthenticationExample[28329:387920] [Presentation] Attempt to present <UIAlertController: 0x149177400> on <_TtGC7SwiftUI19UIHostingControllerGVS_15ModifiedContentVS_7AnyViewVS_12RootModifier__: 0x149027800> (from <UIViewController: 0x13f228090>) which is already presenting <UIAlertController: 0x13d812800>.
```
![Simulator Screen Shot - iPhone 14 Pro - 2023-01-23 at 14 45 53](https://user-images.githubusercontent.com/4164028/214168460-da8cdf0b-6f9b-41ea-9313-05370295ab72.png)

Logged an [issue](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/issues/248) to look into this.

Reply:
So the sleep isn't needed? If not, we should remove it.

Reply:
That was added probably because someone was seeing the error. So I kept as it is and added information in the [issue](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/issues/248) to look into and remove if not needed.